src/entity.rs

```diff
@@ -428,7 +426,7 @@ mod tests {
             },
         );
 
-        let row = vec!["a", "bb", "ccc ddd", "eeee"];
+        let row = vec![smallvec![String::from("a")], smallvec![String::from("bb")], smallvec![String::from("ccc"), String::from("ddd")], smallvec![String::from("eeee")]];
```

Comment:
This is one thing I'm not happy about. It should be possible to keep something much closer to the original signature so this isn't required.

Reply:
There points should be taken into account:
- avoiding creating Strings every new line,
- using as much as possible stack structures (so I think smallvec is ok).